src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -25,12 +30,13 @@ public class JwtTokenReader : IJwtTokenReader
         ///     Constructor
         /// </summary>
         /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
-        /// <param name="applicationId">Azure AD Application used as audience to validate against</param>
-        public JwtTokenReader(string applicationId)
+        /// <param name="claimCheck">Azure AD Application used as audience to validate against</param>
+        public JwtTokenReader(IDictionary<string, string> claimCheck)
         {
-            Guard.NotNullOrWhitespace(applicationId, nameof(applicationId));
+            Guard.NotAny(claimCheck, nameof(claimCheck));
+
+            _claimCheck = claimCheck;
 
```

Comment:
Don't we keep the existing constructor for backwards compatibility ?
The existing constructor which takes the `applicationId` can be implemented as such that it calls the JwtTokenReader claim that accepts the dictionary.

Something like this:

```csharp
public JwtTokenReader( string applicationId ) : this(new Dictionary { "aud" = applicationId })
{}
```
(Syntax errors are possible, but I think you get the idea :) )

Reply:
Yes, absolutly necessary!

Reply:
done